python/vowpalwabbit/dftovw.py

```diff
@@ -753,18 +803,20 @@ def from_colnames(
                 )
             )
 
-        y = y if isinstance(y, list) else [y]
-
-        if label_type not in ["multi_label"]:
-            if len(y) > 1:
-                raise TypeError(
-                    "When label_type is 'simple_label' or 'multiclass', argument 'y' should be a string (or any hashable type) "
-                    + "or a list of exactly one string (or any hashable type)."
-                )
-            else:
-                y = y[0]
+        if label_type and y:
+            y = y if isinstance(y, list) else [y]
+            if label_type not in ["multi_label"]:
+                if len(y) > 1:
+                    raise TypeError(
+                        "When label_type is 'simple_label' or 'multiclass', argument 'y' should be a string (or any hashable type) "
+                        + "or a list of exactly one string (or any hashable type)."
+                    )
+                else:
+                    y = y[0]
 
-        label = dict_label_type[label_type](y)
+            label = dict_label_type[label_type](y)
+        else:
+            label = None
 
         x = x if isinstance(x, list) else [x]
 
```

Comment:
PS: testing for truthy for label_type (`if label_type`) was not necessary since it can't be empty at that point (it would have failed before if it was)